### PR TITLE
Adding states to Norbert network server side

### PR DESCRIPTION
**Pull request by ruiwang-linkedin**
...orbert can set the capability state via API; norbert keeps track of the capability states of its nodes; Upon startup, norbert takes an optional capability state value to set the state; unit tests

### CODE REVIEW

File: cluster/src/main/scala/com/linkedin/norbert/cluster/ClusterClient.scala [hunk 1]
```diff
@@ -192,6 +199,16 @@ trait ClusterClient extends Logging {
   }
 
   /**
+   * Set the capability state for a cluster node
+   * @param nodeId the id of the node to represent capablity
+   *
+   * @throws ClusterDisconnectedException thrown if the clsuter is disconnected when the method is called
+   */
+  def setNodeCapability(nodeId: Int, capability: Long): Unit = handleClusterManagerResponse {
```
Comment (jhartman) [inline cluster/src/main/scala/com/linkedin/norbert/cluster/ClusterClient.scala:211]: Could you javadoc the methods so clients can understand what a capability bit-vector means?

File: cluster/src/main/scala/com/linkedin/norbert/cluster/Node.scala [hunk 2]
```diff
@@ -82,5 +83,5 @@ final case class Node(id: Int, url: String, available: Boolean, partitionIds: Se
     case _ => false
   }
 
-  override def toString = "Node(%d,%s,[%s],%b)".format(id, url, partitionIds.mkString(","), available)
+  override def toString = "Node(%d,%s,[%s],%b,%d)".format(id, url, partitionIds.mkString(","), available, if (capability.isEmpty) -1L else capability.get)
```
Comment (jhartman) [inline cluster/src/main/scala/com/linkedin/norbert/cluster/Node.scala]: Why -1? That is 0xFFFFFFFF in 2-complement, right? Also, should we print the capability in hex or binary? It's meaning is not clear in base 10.

**Comment by jhartman** [comment]
Looks good to me. I was wondering if we could have two helper methods:
addCapability(long capability) and removeCapability(long capability).

The helper methods would take the current capability, and OR or AND it as the methods suggest.

File: cluster/src/main/scala/com/linkedin/norbert/cluster/Node.scala (hunk 2 above)
Comment (ruiwang-linkedin) [inline cluster/src/main/scala/com/linkedin/norbert/cluster/Node.scala]: Great catch. I have made fixed both counts.

**Comment by ruiwang-linkedin** [comment]
I agree it makes sense to have addCapability(Long capacity) and removeCapability(Long capability), but I felt that they might be redundant with setCapability(Long capability). That's why I opt to only implement set. I was trying to avoid too many different flavor of capability related handlings in cluster client.

**Pull request merged**